File: pd-vm/examples/example_complex.lua
local _host = require("pd-vm-host")
local string = require("../stdlib/rss/strings.rss")

-- Complex Lua flavor example: loop + stdlib + host + closure.
local total = 0
for i = 0, 3, 1 do
    total = total + i
end

if string.non_empty("lua") then
    total = add_one(total)
else
    total = 0
end

local base = 7
local add = function(value) return value + base end
base = 8
local closure_value = add(5)

print(closure_value)

File: pd-vm/examples/example_complex.js
import * as string from "../stdlib/rss/strings.rss";
import { add_one } from "pd-vm-host";

// Complex JavaScript flavor example: loop + stdlib + host + closure.
let total = 0;
for (let i = 0; i < 4; i = i + 1) {
    total = total + i;
}

if (!string.non_empty("javascript")) {
    total = 0;
} else {
    total = add_one(total);
}

let base = 7;
let add = (value) => value + base;
base = 8;
let closureValue = add(5);

console.log(closureValue);
